RFC-0001-Agile Coretime.md

```diff
@@ -0,0 +1,291 @@
+
+If the `RegionRecord` value for an elapsed `RegionId` has an `allocation` of `None`, then the item is not removed and the Relay-chain is instructed to place the core for instantaneous use.
+
+### Specific functions of the Broker-chain
+
+Several functions of the Broker-chain SHALL be exposed through dispatchables and/or a `nonfungible` trait implementation integrated into XCM:
+
+#### 1. Transfer of ownership
+
+A `transfer(region: RegionId, new_owner: AccountId)` dispatchable SHALL have the effect of altering the current `owner` field of `region` in the `Regions` map from the signed origin to `new_owner`.
+
+An implementation of the `nonfungible` trait SHOULD include equivalent functionality. `RegionId` SHOULD be used for the `AssetInstance` value.
+
+In both cases, the `price` field SHALL become `None`.
+
+#### 2. Split into segments
+
+A `split(region: RegionId, pivot: Timeslice)` dispatchable SHALL have the effect of mutating the Regions entry of key `region` so that the `end` field becomes `pivot` and create a new Regions entry of the same `core` but a `begin` equal to `pivot` whose `RegionRecord` has the same `owner`, `end` and `allocation` fields as the origin value of `region`.
+
+`price` in both records is/becomes `None`.
+
+Also:
+- `owner` field of `region` must the equal to the Signed origin.
+- `pivot` must equal neither the `begin` nor `end` fields of the `region`.
+
+#### 3. Consumed in exchange for allocation
+
+A dispatchable `allocate(region: RegionId, target: Vec<ParaId>)` SHALL be provided corresponding to the `allocate` function described above.
+
+It MUST be called with a Signed origin equal to the `owner` field of the value of the Regions map for the `region` key. The `allocation` field of this value MUST be `None` (a region may not be re-allocated).
+
+On success, the `allocation` value is changed to `Some` whose inner value is equal to `target`.
+
+If the `begin` field of the `region` parameter is less than the current `Timeslice` value, then it is removed and re-entered with the current `Timeslice` value plus one, unless this would be equal to or greater than the `end` field of the corresponding `RegionRecord` value.
+
+Initially `target` values with only one item MAY be supported.
+
+If the `RegionRecord`'s `price` field is `Some` (indicating that the Region is freshly purchased), then the Broker-chain SHALL record an entry in a storage map called AllowedRenewals. This maps a `CoreIndex` to a struct `RenewalRecord`:
+
+```rust
+struct RenewalRecord {
+    target: Vec<ParaId>,
+    price: Balance,
+    sale: SaleIndex,
+}
+```
+
+#### 4. Renewals
+
+A dispatchable `renew(core: CoreIndex)` SHALL be provided. Any account may call `renew` to purchase Bulk Coretime and renew an active allocation for the given `core`.
+
+This MUST be called during the `LEADIN_PERIOD` prior to a Bulk sale (exactly like `purchase`) and has the same effect as `purchase` followed by `allocate` containing the same `Vec<ParaId>`, except that:
```

Comment:
If we have a region `(start, end, owner, price, allocation: None)`, and `renew` isn't called, what happens to it? Let's say it's e.g. the last `7 * DAYS`. The first `21 * DAYS` part of it is renewed by its owner, and this is not. Who gets the 7-day chunk that wasn't renewed?

Reply:
`renew` can only be called on regions which were the full `BULK_PERIOD` in length and the assignment but be exactly the same from month to month.